metomi/rose/config_processors/fileinstall.py

```diff
@@ -260,23 +260,20 @@ def _process(self, conf_tree, nodes, loc_dao, **kwargs):
                 # See if any sources have changed names.
                 if not target.is_out_of_date:
                     conn = loc_dao.get_conn()
-                    try:
-                        prev_dep_locs = conn.execute(
-                            "SELECT * FROM dep_names WHERE name=?",
-                            [target.name]
-                        ).fetchall()
-                        prev_dep_locs = [i[1] for i in prev_dep_locs]
-                        prev_dep_locs = [
-                            loc_dao.select(i)
-                            for i in prev_dep_locs
-                        ]
-                        if (
-                            [i.name for i in prev_dep_locs] !=
-                            [i.name for i in target.dep_locs]
-                        ):
-                            target.is_out_of_date = True
-                    finally:
-                        conn.close()
```

Comment:
When does it get closed, then?

Reply:
This got put here because of a review comment by me on another branch but it turned out to be completely wrong. I don't know where or if the connection is closed (other than by an implicit try/finally in sys.exit or something like that) would have to go digging.